### DISCUSSION TITLE
Does CAS style transaction require DoMulti(...)

### DISCUSSION THREAD

**Discussion by tm1303** [redis/rueidis#494] [Q&A] [answered]
The given example in the README is as follows (modified very slightly to return err, i know the ToAny() is wrong 😝)
```
client.Dedicated(func(c rueidis.DedicatedClient) error {
    // watch keys first
    c.Do(ctx, c.B().Watch().Key("k1", "k2").Build())
    // perform read here
    c.Do(ctx, c.B().Mget().Key("k1", "k2").Build())
    // perform write with MULTI EXEC
    _, err := c.DoMulti(
        ctx,
        c.B().Multi().Build(),
        c.B().Set().Key("k1").Value("1").Build(),
        c.B().Set().Key("k2").Value("2").Build(),
        c.B().Exec().Build(),
    ).ToAny()
    return err
})
```
I only have a single `Set()` to perform but I'm unclear if I must pass my one `Set()` as param to `DoMulti()` to benefit from the CAS style transaction or if this example is just showing us how to (optionally) perform multiple `Set()` within a CAS transaction.

I have one `Set()` to perform but it is conditional on the watched key not changing. Would the following suffice? Would this `Set()` fail and return an appropriate error if the watched key value changes or expires?
```

client.Dedicated(func(c rueidis.DedicatedClient) error {
    // watch keys first
    c.Do(ctx, c.B().Watch().Key("k1").Build())
    // perform read here
    c.Do(ctx, c.B().Get().Key("k1").Build())
    // perform single write
    _, err := c.Do(ctx, c.B().Set().Key("k1").Value("1").Build()).ToAny()

    return err
})
```

**Comment by rueian** [comment, marked as answer]
Hi @tm1303,

If you rely on the WATCH, then there are two must:
1. Using the WATCH in a dedicated connection.
2. Modifying watched keys in a MULTI EXEC block in the same connection.

DoMulti is not required, but generally recommended.

**Reply by tm1303** [reply]
10-4 good buddy, thanks for coming back to me now y'all.
